apps/web-giddh/src/app/permissions/components/list/permission-list.html

```diff
@@ -6,26 +6,42 @@
 >
     <hamburger-menu [pageHeading]="localeData?.page_heading"></hamburger-menu>
     <div class="container-fluid clearfix" *ngIf="allRoles?.length > 0">
-        <a
-            href="javascript:void(0)"
-            *ngIf="showBackButton"
-            class="btn btn-sky-blue pull-right mr-t1 permission-btn mr-b1"
-            [routerLink]="['/pages/settings/permission']"
-            >{{commonLocaleData?.app_back}}</a
-        >
-        <div class="col-sm-12 mr-t15 p-0 role clearfix" *ngFor="let role of allRoles">
+        <div class="col-sm-12 mr-t2 mr-b2 d-flex justify-content-end column-gap5">
+            <button mat-stroked-button color="primary" (click)="openPermissionDialog()">
+                {{localeData?.add_new_role}}
+            </button>
+            <button
+                mat-stroked-button
+                *ngIf="showBackButton"
+                class="btn custom-btn"
+                [routerLink]="['/pages/settings/permission']"
+            >
+                {{commonLocaleData?.app_back}}
+            </button>
+        </div>
+        <div class="col-sm-12 mr-t15 p-0 role clearfix btn-show" *ngFor="let role of allRoles">
             <div class="row mr-0 ml-0">
                 <div class="col-md-3 col-sm-3 col-12 p-0 role-head cp">
                     <span class="vertical-center">
                         <h1>{{role.name}}</h1>
-                        <div class="text-center mr-t2 on-hover">
-                            <button *ngIf="!role.isFixed" class="btn btn-sm btn-primary" (click)="updateRole(role);">
+                        <div class="text-center mr-t2 on-hover d-flex column-gap1">
+                            <button
+                                mat-stroked-button
+                                *ngIf="!role.isFixed"
```

Comment:
```suggestion
                                *ngIf="!role?.isFixed"
```
change in over all file